R/target_market_share.R

```diff
@@ -98,45 +98,9 @@ target_market_share <- function(data,
 
   data <- ungroup(warn_grouped(data, "Ungrouping input data."))
 
-  valid_columns <- c(
-    "id_loan",
-    "id_direct_loantaker",
-    "name_direct_loantaker",
-    "id_intermediate_parent_1",
-    "name_intermediate_parent_1",
-    "id_ultimate_parent",
-    "name_ultimate_parent",
-    "loan_size_outstanding",
-    "loan_size_outstanding_currency",
-    "loan_size_credit_limit",
-    "loan_size_credit_limit_currency",
-    "sector_classification_system",
-    "sector_classification_input_type",
-    "sector_classification_direct_loantaker",
-    "fi_type",
-    "flag_project_finance_loan",
-    "name_project",
-    "lei_direct_loantaker",
-    "isin_direct_loantaker",
-    "id_2dii",
-    "level",
-    "sector",
-    "sector_ald",
-    "name",
-    "name_ald",
-    "score",
-    "source",
-    "borderline"
-  )
+  check_input_for_crucial_columns(data, ald, scenario)
 
-  check_valid_columns(data, valid_columns)
-
-  crucial_scenario <- c("scenario", "tmsr", "smsp")
-  check_crucial_names(scenario, crucial_scenario)
-  check_crucial_names(ald, "is_ultimate_owner")
-  walk_(crucial_scenario, ~ check_no_value_is_missing(scenario, .x))
-
-  data <- aggregate_by_loan_id(data)
+  data <- aggregate_by_name_ald(data)
```

Comment:
Yes, but this will have to come in a future PR. 
The ALD->ABCD attack is going to be swift and focused. I'd rather not start changing it in some places, but not other, i think that will cause confusion.